src/Middleware/SpaServices.Extensions/src/AngularCli/AngularCliMiddleware.cs

```diff
@@ -62,9 +62,17 @@ internal static class AngularCliMiddleware
         }
 
         private static async Task<AngularCliServerInfo> StartAngularCliServerAsync(
-            string sourcePath, string npmScriptName, ILogger logger)
+            string sourcePath, string npmScriptName, int portNumber, ILogger logger)
         {
-            var portNumber = TcpPortFinder.FindAvailablePort();
+            if (portNumber == 0) {
+                portNumber = TcpPortFinder.FindAvailablePort();
+            } else {
+                if (!TcpPortFinder.PortAvailable(portNumber)) {
```

Comment:
If the developer specified a port explicitly but that port is not available, I think it would be better just to throw and fail rather than trying to pick another port.

This is because, if the developer really wants it to be on a specific port, they probably have a reason for this and something else in their infrastructure will depend on it actually being on that port.

So I recommend removing `PortAvailable` altogether and letting it just fail if the specified port can't be bound.

Reply:
Yes, I agree. I've removed that part again. Let me know if there's more I can do. :-)